### PR TITLE
Fixing anchors and links in documentation

### PR DESCRIPTION
**Pull request by ahus1**
This PR fixed link anchors and cross references in the documentation.

In order for a link to work:
* the file needs to exist
* anchors are case sensitive
* ~~local references within a file should start with #~~
* the idprefix/idseparator are set up like on the live site of Quarkus (that is no prefixed underscore)

### CODE REVIEW

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 1]
```diff
@@ -21,8 +21,8 @@ To complete this guide, you need:
 * less than 15 minutes
 * an IDE
 * JDK 8 installed with `JAVA_HOME` configured appropriately
-* A xref:configuring-c-development[working C development environment]
-* GraalVM version {graalvm-version} installed and xref:configuring-graalvm[configured appropriately]
+* A xref:#configuring-c-development[working C development environment]
+* GraalVM version {graalvm-version} installed and xref:#configuring-graalvm[configured appropriately]
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc]: Hmmm, these links are working correctly on the website. Is it only an issue when you generate locally?

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 2]
```diff
@@ -90,7 +90,7 @@ Installing via scoop will do this for you.
 ${GRAALVM_HOME}/bin/gu install native-image
 ----
 +
-Some previous releases of GraalVM included the `native-image` tool by default.  This is no longer the case; it must be installed as a second step after GraalVM itself is installed. Note: there is an outstanding issue xref:graal-and-catalina[using GraalVM with macOS Catalina].
+Some previous releases of GraalVM included the `native-image` tool by default.  This is no longer the case; it must be installed as a second step after GraalVM itself is installed. Note: there is an outstanding issue xref:#graal-and-catalina[using GraalVM with macOS Catalina].
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc]: Same here.

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 3]
```diff
@@ -537,7 +537,7 @@ If you need to generate resources for both platforms (vanilla Kubernetes and Ope
 quarkus.kubernetes.deployment-target=kubernetes, openshift
 ----
 
-NOTE: Quarkus also provides the link:openshift[OpenShift] extension. This extension is basically wrapper around the Kubernetes extension and
+NOTE: Quarkus also provides the link:#openshift[OpenShift] extension. This extension is basically wrapper around the Kubernetes extension and
```
Comment (gsmet) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc]:
```suggestion
NOTE: Quarkus also provides the link:deploying-to-openshift[OpenShift] extension. This extension is basically a wrapper around the Kubernetes extension and
```

**Review by gsmet — CHANGES_REQUESTED** [review]
I added 1 question and 1 suggestion.

Thanks a lot for taking the time to prepare this PR!

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 1 above)
Comment (ahus1) [reply docs/src/main/asciidoc/building-native-image.adoc]: This is probably only my personal convention, the # is optional. I'll update the PR.

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 2 above)
Comment (ahus1) [reply docs/src/main/asciidoc/building-native-image.adoc]: This is probably only my personal convention, the # is optional. I'll update the PR.

**Comment by ahus1** [comment]
@gsmet - thank you for your reviews. I've update the changes accordingly and squashed/force-pushed the result. Ready for another review.

**Review by gsmet — APPROVED** [review]
Perfect, thanks!

**Pull request merged**